Fix Reference Cleaner

hjpalpha opened the pull request:
## Summary
same issue as with #848

## How did you test this change?
not tested, but the same change was tested in a different module for #848

JaspervanRiet commented:
Also not severe, I'll add a definition for that (along with other documentation) soon. This has been live for months, thus it's simply not severe.

JaspervanRiet commented:
Clear case to add tests.

JaspervanRiet commented:
I added a test but it's currently working just fine, _without this change_. So please add a test case that is failing.

Rathoz commented on standard/test/reference_cleaner_test.lua:
```diff
@@ -1,3 +1,11 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Class
```
Wrong module

JaspervanRiet replied:
Fixed

hjpalpha commented:
> I added a test but it's currently working just fine, _without this change_. So please add a test case that is failing.

it only does not fail because it cuts off the 2nd `-01` again (is matches `%d+-%d+-%d+`), but if you have a `-??` in the middle, e.g. (`2011-??-05`) it will fail, added it to the module accordingly

JaspervanRiet reviewed (approved):
LGTM
